source/material_model/diffusion_dislocation.cc

```diff
@@ -0,0 +1,527 @@
+          AssertThrow(x_values.size() == 1u || (x_values.size() == n_fields),
+                      ExcMessage("Length of activation volume for diffusion list must be either one, or n_compositional_fields+1"));
+          if (x_values.size() == 1)
+            activation_volumes_diffusion.assign( n_fields , x_values[0] );
+          else
+            activation_volumes_diffusion = x_values;
+
+
+          // Dislocation creep parameters
+          // ---- dislocation creep prefactors
+          x_values = Utilities::string_to_double(Utilities::split_string_list(prm.get ("Prefactors for dislocation creep")));
+          AssertThrow(x_values.size() == 1u || (x_values.size() == n_fields),
+                      ExcMessage("Length of prefactors for dislocation list must be either one, or n_compositional_fields+1"));
+          if (x_values.size() == 1)
+            prefactors_dislocation.assign( n_fields , x_values[0] );
+          else
+            prefactors_dislocation = x_values;
+
+          // ---- dislocation creep activation energies
+          x_values = Utilities::string_to_double(Utilities::split_string_list(prm.get ("Activation energies for dislocation creep")));
+          AssertThrow(x_values.size() == 1u || (x_values.size() == n_fields),
+                      ExcMessage("Length of activation energy for dislocation list must be either one, or n_compositional_fields+1"));
+          if (x_values.size() == 1)
+            activation_energies_dislocation.assign( n_fields , x_values[0] );
+          else
+            activation_energies_dislocation = x_values;
+
+          // ---- dislocation creep activation volumes
+          x_values = Utilities::string_to_double(Utilities::split_string_list(prm.get ("Activation volumes for dislocation creep")));
+          AssertThrow(x_values.size() == 1u || (x_values.size() == n_fields),
+                      ExcMessage("Length of activation volume for dislocation list must be either one, or n_compositional_fields+1"));
+          if (x_values.size() == 1)
+            activation_volumes_dislocation.assign( n_fields , x_values[0] );
+          else
+            activation_volumes_dislocation = x_values;
+
+          // ---- dislocation creep stress exponents
+          x_values = Utilities::string_to_double(Utilities::split_string_list(prm.get ("Stress exponents for dislocation creep")));
+          AssertThrow(x_values.size() == 1u || (x_values.size() == n_fields),
+                      ExcMessage("Length of stress exponents for dislocation list must be either one, or n_compositional_fields+1"));
+          if (x_values.size() == 1)
+            stress_exponents_dislocation.assign( n_fields , x_values[0] );
+          else
+            stress_exponents_dislocation = x_values;
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+    ASPECT_REGISTER_MATERIAL_MODEL(Diffusion_Dislocation,
+                                   " Diffusion_Dislocation",
```

Comment:
use "diffusion dislocation" or "diffusion-dislocation" or something similar for this line. We usually use full words, lower case, including spaces for the names of models.